io/src/main/scala/sbt/internal/io/Milli.scala

```diff
@@ -281,12 +290,16 @@ private object WinMilli extends MilliNative[FILETIME] {
                            null)
     if (hFile == INVALID_HANDLE_VALUE) {
       val err = GetLastError()
-      if (err == ERROR_FILE_NOT_FOUND || err == ERROR_PATH_NOT_FOUND)
+      if (err == ERROR_FILE_NOT_FOUND || err == ERROR_PATH_NOT_FOUND || err == ERROR_ACCESS_DENIED)
         throw new FileNotFoundException("Not found: " + lpFileName)
-      else
+      else if (err == ERROR_SHARING_VIOLATION && attempt < MAX_GET_HANDLE_ATTEMPTS) {
+        Thread.sleep(1)
+        getHandle(lpFileName, dwDesiredAccess, dwShareMode, attempt + 1)
```

Comment:
The `ERROR_SHARING_VIOLATION` should (in theory) already be fixed by this pull request: https://github.com/sbt/io/pull/134
That PR was merged in 1.x (on March 7th) but it is not in 1.1.x. The 1.1.x branch is still very much active at this point, so it may be worth backporting it, rather than using workarounds. /cc @dwijnand

Reply:
In https://github.com/sbt/sbt/issues/3972#issuecomment-384634235, a user reports that an `ERROR_SHARING_VIOLATION` still appears in a 1.x snapshot when running under the Windows Subsystem for Linux (with #134 applied); conversely in https://github.com/sbt/sbt/issues/3972#issuecomment-369457527 the patch seems to have resolved the issue.
Access permissions are weird in Windows, and I was unable to reproduce it post-patch. In the context of this pull request, I would start by backporting #134 to 1.1.x, and use the workaround proposed by @eatkins only if that is still a problem in Appveyor afterwards. That would be my suggestion, at least.